actioncable/lib/action_cable/connection/base.rb

```diff
@@ -186,7 +191,10 @@ def handle_close
           subscriptions.unsubscribe_from_all
           unsubscribe_from_internal_channel
 
-          disconnect if respond_to?(:disconnect)
+          payload = { connection_class: self.class.name, request: request }
+          ActiveSupport::Notifications.instrument("disconnect.action_cable", payload) do
```

Comment:
Does it make sense to instrument disconnect if there's no `disconnect` defined, i.e. the respond to is false?

Reply:
Let's also extract an either:

```
instrument :disconnect do
  disconnect if respond_to?(:disconnect)
end

# Or:
instrument(:disconnect) { disconnect } if respond_to?(:disconnect)
```

(Put the method below `successful_request_message`)